BUG: Windows Built SciPy can't import _fblas via pip install .

### Describe your issue.

After building SciPy main branch, installing it systemwide via `pip install .` leads to broken import. This happens even building NumPy locally with the locally built OpenBLAS. 

```python
In [6]: np.show_config()
blas_armpl_info:
  NOT AVAILABLE
blas_mkl_info:
  NOT AVAILABLE
blis_info:
  NOT AVAILABLE
openblas_info:
    library_dirs = ['C:\\Users\\ilhan\\Documents\\GitHub\\numpy\\build\\openblas_info']
    libraries = ['openblas_info']
    language = f77
    define_macros = [('HAVE_CBLAS', None)]
blas_opt_info:
    library_dirs = ['C:\\Users\\ilhan\\Documents\\GitHub\\numpy\\build\\openblas_info']
    libraries = ['openblas_info']
    language = f77
    define_macros = [('HAVE_CBLAS', None)]
lapack_armpl_info:
  NOT AVAILABLE
lapack_mkl_info:
  NOT AVAILABLE
openblas_lapack_info:
    library_dirs = ['C:\\Users\\ilhan\\Documents\\GitHub\\numpy\\build\\openblas_lapack_info']
    libraries = ['openblas_lapack_info']
    language = f77
    define_macros = [('HAVE_CBLAS', None)]
lapack_opt_info:
    library_dirs = ['C:\\Users\\ilhan\\Documents\\GitHub\\numpy\\build\\openblas_lapack_info']
    libraries = ['openblas_lapack_info']
    language = f77
    define_macros = [('HAVE_CBLAS', None)]
Supported SIMD extensions in this NumPy install:
    baseline = SSE,SSE2,SSE3
    found = SSSE3,SSE41,POPCNT,SSE42,AVX,F16C,FMA3,AVX2,AVX512F,AVX512CD,AVX512_SKX,AVX512_CLX,AVX512_CNL,AVX512_ICL
    not found =
```

### Reproducing Code Example

```python
C:\Users\ilhan\Documents\GitHub\scipy [main ≡]> $env:PKG_CONFIG_PATH="C:\opt\64\lib\pkgconfig\"
C:\Users\ilhan\Documents\GitHub\scipy [main ≡]> $env:CC="x86_64-w64-mingw32-gcc";$env:CXX="x86_64-w64-mingw32-g++"
C:\Users\ilhan\Documents\GitHub\scipy [main ≡]> pip install .
Processing c:\users\ilhan\documents\github\scipy
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: numpy>=1.19.5 in c:\users\ilhan\appdata\local\programs\python\python310\lib\site-packages\numpy-1.24.0.dev0+695.g50a74fb65-py3.10-win-amd64.egg (from scipy==1.10.0.dev0) (1.24.0.dev0+695.g50a74fb65)
Building wheels for collected packages: scipy
  Building wheel for scipy (pyproject.toml) ... done
  Created wheel for scipy: filename=scipy-1.10.0.dev0-cp310-cp310-win_amd64.whl size=47000726 sha256=35808c560f2e33dafe3ced612713c8565d77b105aec693124ae4d13cde1afdb8
  Stored in directory: C:\Users\ilhan\AppData\Local\Temp\pip-ephem-wheel-cache-04ckr_zh\wheels\63\13\46\e3e2e5525b45f526ecce5f57def1f4d2068c8fc759a4aae4eb
Successfully built scipy
Installing collected packages: scipy
Successfully installed scipy-1.10.0.dev0
```


### Error message

```shell
C:\Users\ilhan\Documents> ipython
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import scipy.linalg as la
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Input In [1], in <cell line: 1>()
----> 1 import scipy.linalg as la

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\scipy\linalg\__init__.py:197, in <module>
      1 """
      2 ====================================
      3 Linear algebra (:mod:`scipy.linalg`)
   (...)
    194
    195 """  # noqa: E501
--> 197 from ._misc import *
    198 from ._cythonized_array_utils import *
    199 from ._basic import *

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\scipy\linalg\_misc.py:3, in <module>
      1 import numpy as np
      2 from numpy.linalg import LinAlgError
----> 3 from .blas import get_blas_funcs
      4 from .lapack import get_lapack_funcs
      6 __all__ = ['LinAlgError', 'LinAlgWarning', 'norm']

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\scipy\linalg\blas.py:213, in <module>
    210 import numpy as _np
    211 import functools
--> 213 from scipy.linalg import _fblas
    214 try:
    215     from scipy.linalg import _cblas

ImportError: DLL load failed while importing _fblas: The specified module could not be found.
```


### SciPy/NumPy/Python version information

main, main, '3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]'

This is all I get from the compilation and I don't get any such messages. I'll hack into dev.py to print the full paths when I have time. 

```bash
C:\Users\ilhan\Documents\GitHub\scipy [main ≡]> $env:CC="x86_64-w64-mingw32-gcc";$env:CXX="x86_64-w64-mingw32-g++"
C:\Users\ilhan\Documents\GitHub\scipy [main ≡]> $env:PKG_CONFIG_PATH="C:\opt\64\lib\pkgconfig\"
C:\Users\ilhan\Documents\GitHub\scipy [main ≡]> python dev.py build --win-cp-openblas
💻  meson setup C:\Users\ilhan\Documents\GitHub\scipy\build --prefix C:\Users\ilhan\Documents\GitHub\scipy\build-installThe Meson build system
Version: 0.63.1
Source dir: C:\Users\ilhan\Documents\GitHub\scipy
Build dir: C:\Users\ilhan\Documents\GitHub\scipy\build
Build type: native build
Project name: SciPy
Project version: 1.10.0.dev0
C compiler for the host machine: x86_64-w64-mingw32-gcc (gcc 12.2.0 "x86_64-w64-mingw32-gcc (Rev1, Built by MSYS2 project) 12.2.0")
C linker for the host machine: x86_64-w64-mingw32-gcc ld.bfd 2.39
C++ compiler for the host machine: x86_64-w64-mingw32-g++ (gcc 12.2.0 "x86_64-w64-mingw32-g++ (Rev1, Built by MSYS2 project) 12.2.0")
C++ linker for the host machine: x86_64-w64-mingw32-g++ ld.bfd 2.39
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C supports arguments -Wno-unused-but-set-variable: YES
Library m found: YES
Fortran compiler for the host machine: gfortran (gcc 12.2.0 "GNU Fortran (Rev1, Built by MSYS2 project) 12.2.0")
Fortran linker for the host machine: gfortran ld.bfd 2.39
Program cython found: YES (C:\Users\ilhan\AppData\Local\Programs\Python\Python310\Scripts\cython.EXE)
Program pythran found: YES (C:\Users\ilhan\AppData\Local\Programs\Python\Python310\Scripts\pythran.EXE)
Program scipy/_build_utils/copyfiles.py found: YES (C:\Users\ilhan\AppData\Local\Programs\Python\Python310\python.exe C:\Users\ilhan\Documents\GitHub\scipy\scipy/_build_utils/copyfiles.py)
Program python3 found: YES (C:\Users\ilhan\AppData\Local\Programs\Python\Python310\python.exe)
WARNING: You should add the boolean check kwarg to the run_command call.
         It currently defaults to false,
         but it will default to true in future releases of meson.
         See also: https://github.com/mesonbuild/meson/issues/9300
Library npymath found: YES
Library npyrandom found: YES
Found pkg-config: C:\msys64\ucrt64\bin\pkg-config.EXE (1.8.0)
Run-time dependency openblas found: YES 0.3.21
Dependency openblas found: YES 0.3.21 (cached)
Program _build_utils/cythoner.py found: YES (C:\Users\ilhan\AppData\Local\Programs\Python\Python310\python.exe C:\Users\ilhan\Documents\GitHub\scipy\scipy\_build_utils/cythoner.py)
Checking for function "open_memstream" : NO
Configuring messagestream_config.h using configuration
Checking for size of "void*" : 8
Run-time dependency threads found: YES
Checking for size of "void*" : 8
Dependency threads found: YES unknown (cached)
Compiler for C supports arguments -Wno-unused-but-set-variable: YES (cached)
Compiler for C supports arguments -Wno-unused-but-set-variable: YES (cached)
Compiler for C++ supports arguments -Wno-format-truncation: YES
Compiler for C++ supports arguments -Wno-class-memaccess: YES
Build targets in project: 206

SciPy 1.10.0.dev0

  User defined options
    prefix: C:\Users\ilhan\Documents\GitHub\scipy\build-install

Found ninja-1.10.2.git.kitware.jobserver-1 at C:\Users\ilhan\AppData\Local\Programs\Python\Python310\Scripts\ninja.EXE
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_ufuncs_pxi_pxd' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_ufuncs_pxi_pxd' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_ufuncs_pxi_pxd' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_ufuncs_pxi_pxd' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_ufuncs_pxi_pxd' has more than one output! Using the first one.
WARNING: custom_target '_stats_pxd' has more than one output! Using the first one.
WARNING: custom_target 'cython_special' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_spatial_pxd' has more than one output! Using the first one.
WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_spatial_pxd' has more than one output! Using the first one.
WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_dummy_init_cyoptimize' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
Meson build setup OK
💻  ninja -C C:\Users\ilhan\Documents\GitHub\scipy\build
ninja: Entering directory `C:\Users\ilhan\Documents\GitHub\scipy\build'
[123/1562] Compiling C object scipy/linalg/_matfuncs_sqrtm_triu.cp310-win_amd64.pyd.p/meson-generated__matfuncs_sqrtm_triu.c.obj
In file included from C:\Users\ilhan\AppData\Local\Programs\Python\Python310\lib\site-packages\numpy-1.24.0.dev0+695.g50a74fb65-py3.10-win-amd64.egg\numpy\core\include/numpy/ndarraytypes.h:1940,
                 from C:\Users\ilhan\AppData\Local\Programs\Python\Python310\lib\site-packages\numpy-1.24.0.dev0+695.g50a74fb65-py3.10-win-amd64.egg\numpy\core\include/numpy/ndarrayobject.h:12,
                 from C:\Users\ilhan\AppData\Local\Programs\Python\Python310\lib\site-packages\numpy-1.24.0.dev0+695.g50a74fb65-py3.10-win-amd64.egg\numpy\core\include/numpy/arrayobject.h:5,
                 from scipy/linalg/_matfuncs_sqrtm_triu.cp310-win_amd64.pyd.p/_matfuncs_sqrtm_triu.c:746:
C:\Users\ilhan\AppData\Local\Programs\Python\Python310\lib\site-packages\numpy-1.24.0.dev0+695.g50a74fb65-py3.10-win-amd64.egg\numpy\core\include/numpy/npy_1_7_deprecated_api.h:14:9: note: '#pragma message: C:\Users\ilhan\AppData\Local\Programs\Python\Python310\lib\site-packages\numpy-1.24.0.dev0+695.g50a74fb65-py3.10-win-amd64.egg\numpy\core\include/numpy/npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION'
   14 | #pragma message(_WARN___LOC__"Using deprecated NumPy API, disable it with " \
      |         ^~~~~~~
[125/1562] Compiling C object scipy/linalg/_solve_toeplitz.cp310-win_amd64.pyd.p/meson-generated__solve_toeplitz.c.obj


[snipped all the compiler warnings]

NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION'
   15 |                          "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION")
      |                                                                             ^
scipy/optimize/_highs/_highs_wrapper.cp310-win_amd64.pyd.p/_highs_wrapper.cpp: In function 'PyObject* __Pyx_Import(PyObject*, PyObject*, int)':
scipy/optimize/_highs/_highs_wrapper.cp310-win_amd64.pyd.p/_highs_wrapper.cpp:25733:31: warning: comparing the result of pointer addition '(((const char*)"scipy.optimize._highs.cython.src._highs_wrapper") + 5
' and NULL [-Waddress]
25733 |             if ((1) && (strchr(__Pyx_MODULE_NAME, '.'))) {
      |                        ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
[1562/1562] Linking target scipy/optimize/_highs/_highs_wrapper.cp310-win_amd64.pyd
Build OK
💻  meson install -C build
Installing, see meson-install.log...
Installation OK
OpenBLAS copied
C:\Users\ilhan\Documents\GitHub\scipy [main ≡]>
```

--- a/dev.py
+++ b/dev.py
@@ -558,19 +558,39 @@ class Build(Task):
         default `_distributor_init.py` file with the one
         we use for wheels uploaded to PyPI so that DLL gets loaded.
 
-        Assumes pkg-config is installed and aware of OpenBLAS.
+        Assumes pkg-config is installed and aware of OpenBLAS. 
+
+        The "dirs" parameter is typically a "Dirs" object with the
+        structure as the following, say, if dev.py is run from the
+        folder "repo":
+
+        dirs = Dirs(
+            root=WindowsPath('C:/.../repo'),
+            build=WindowsPath('C:/.../repo/build'),
+            installed=WindowsPath('C:/.../repo/build-install'),
+            site=WindowsPath('C:/.../repo/build-install/Lib/site-packages'
+            )
+
         """
         # Get OpenBLAS lib path from pkg-config
         cmd = ['pkg-config', '--variable', 'libdir', 'openblas']
         result = subprocess.run(cmd, capture_output=True, text=True)
+        # pkg-config does not return any meaningful error message if fails
         if result.returncode != 0:
-            print(result.stderrr)
+            print('"pkg-config --variable libdir openblas" '
+                  'command did not manage to find OpenBLAS '
+                  'succesfully. Try running manually on the '
+                  'command prompt for more information.')
             return result.returncode
 
-        openblas_lib_path = Path(result.stdout.strip())
+        # Skip the drive letter of the path -> /c to get Windows drive
+        # to be appended correctly to avoid "C:\c\..." from stdout.
+        openblas_lib_path = Path(result.stdout.strip()[2:]).resolve()
         if not openblas_lib_path.stem == 'lib':
-            raise RuntimeError(
-                f'Expecting "lib" at end of "{openblas_lib_path}"')
+            raise RuntimeError('"pkg-config --variable libdir openblas" '
+                               'command did not return a path ending with'
+                               ' "lib" folder. Instead it returned '
+                               f'"{openblas_lib_path}"')
 
         # Look in bin subdirectory for OpenBLAS binaries.
         bin_path = openblas_lib_path.parent / 'bin'
@@ -580,8 +600,8 @@ class Build(Task):
         libs_path.mkdir(exist_ok=True)
         # Copy DLL files from OpenBLAS install to scipy install .libs subdir.
         for dll_fn in bin_path.glob('*.dll'):
-            out_fname = libs_path / dll_fn.parts[-1]
-            print(f'Copying {dll_fn} to {out_fname}')
+            out_fname = libs_path / dll_fn.name
+            print(f'Copying {dll_fn} ----> {out_fname}')
             out_fname.write_bytes(dll_fn.read_bytes())
 
         # Write _distributor_init.py to scipy install dir;

